foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_8_0/AiopsReferencePythonPropertiesFileMigration.java

```diff
@@ -0,0 +1,119 @@
+
+    public static final Logger logger = LoggerFactory.getLogger(AiopsReferencePythonPropertiesFileMigration.class);
+    public static final Map<String, String> AIOPS_REFERENCE_PYTHON_PROPERTIES_FILE_PACKAGE_MAP = Map.of(
+            "policy-decision-point.aiops-security.properties.vm", "policy-decision-point.aissemble-security.properties.vm"
+            );
+
+    /**
+     * Function to check whether the migration is necessary.
+     * @param file file to check
+     * @return shouldExecute - whether the migration is necessary.
+     */
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        boolean shouldExecute = false;
+
+//        getMavenProject().getPlugin("");
+
+        try {
+            if(file.getName().equals("aiops-security.properties")) {
+                BufferedReader aiopsReferencePythonPropertiesFileConfig = new BufferedReader((new FileReader(file)));
+
+                String lineOne;
+                while((lineOne = aiopsReferencePythonPropertiesFileConfig.readLine()) !=null && !shouldExecute) {
+                    for (String key : AIOPS_REFERENCE_PYTHON_PROPERTIES_FILE_PACKAGE_MAP.keySet()) {
+                        if (lineOne.contains(key)) {
+                            return true;
+                        }
+                    }
+                }
+            } else {
+                BufferedReader aiopsReferencePythonDockerfileConfig = new BufferedReader((new FileReader(file)));
+
+                String lineTwo;
+                while((lineTwo = aiopsReferencePythonDockerfileConfig.readLine()) !=null && !shouldExecute) {
+                    for (String key : AIOPS_REFERENCE_PYTHON_PROPERTIES_FILE_PACKAGE_MAP.keySet()) {
+                        if (lineTwo.contains(key)) {
+                            return true;
+                        }
+                    }
+                }
+            }
+        } catch (Exception e) {
+            logger.error("Error in determining whether a 'AIOPS' renamed properties file requires a migration.");
+            throw new BatonException(e);
+        }
+
+        return shouldExecute;
+    }
+
+    private static void fileNameRefactor(Path oldPath, Path newPath) {
+        try {
+            Files.move(oldPath, newPath);
+        } catch (IOException e) {
+            logger.error("Error in refactoring the file in AiopsReferencePythonPropertiesFileMigration.");
+            throw new BatonException(e);
+        }
+    }
+
+    /**
+     * Performs the migration if the shouldExecuteOnFile() returns true.
+     * @param file file to migrate
+     * @return isMigrated - Whether the file was migrated successfully.
+     */
+    @Override
+    protected boolean performMigration(File file) {
+        boolean isMigrated = false;
+
+        try {
+//            // I am not sure how to rename the file automatically, is this something that could potentially be a part of the manual upgrade instructions?
```

Comment:
can we remove this? or is this a comment for future reference (for the follow on ticket)?

Reply:
I will remove this.